tests/test/CMakeLists.txt

```diff
@@ -1,4 +1,10 @@
 file(GLOB_RECURSE TEST_FILES ${CMAKE_CURRENT_LIST_DIR} test_*.cpp)
+file(GLOB_RECURSE SGX_TEST_FILES sgx test_*.cpp)
+
+# Remove SGX tests if SGX disabled or not found
+if (NOT (FAASM_SGX_ENABLED AND SGX_FOUND))
+    list(REMOVE_ITEM ${TEST_FILES} ${SGX_TEST_FILES})
+endif()
```

Comment:
As per comment above, can't we just rely on `SGX_FOUND`? Why do we need the extra `FAASM_SGX_ENABLED` variable?

Reply:
Ah ok, I would say the normal behaviour is for builds to look for what they need on a system, then build with it if they find it (e.g. something will build with CUDA or BLAS support if it finds that library). Was it failing with your SGX install? I'd rather we keep compiling SGX support by default otherwise the code will rot. 

If you do still want to turn it off, I think the option should be to opt out (rather than opt in), i.e. default to `FAASM_SGX_ENABLED=ON`

Reply:
Ok, I see what you mean. SGX build is also sorted now, so I will revert to the previous behaviour (with the caveat of the removal of `sgx` tests if `NOT SGX_FOUND`.